Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseOpMode.java

```diff
@@ -235,7 +254,7 @@ public void centerJunctionTop(GrabberCameraPipeline pipeline) {
             }
 
         // while the cone isn't centered over the junction
-        } while (Math.abs(xOffset) > Constants.JUNCTION_TOP_TOLERANCE || Math.abs(yOffset) > Constants.JUNCTION_TOP_TOLERANCE);
+        } while (Math.abs(xOffset) > Constants.JUNCTION_TOP_TOLERANCE || Math.abs(yOffset) > Constants.JUNCTION_TOP_TOLERANCE || !cancelled);
```

Comment:
Do you really want this to be "or not canceled"? That means whenever it is not canceled it will do the loop regardless of the other checks here. I think you probably want this. 
```suggestion
        } while ((Math.abs(xOffset) > Constants.JUNCTION_TOP_TOLERANCE || Math.abs(yOffset) > Constants.JUNCTION_TOP_TOLERANCE) && !cancelled);
```

Reply:
Replaced